### PR TITLE
Breaking change note on Task->ValueTask

### PR DESCRIPTION
**Pull request by roji**
Describes changes from PR https://github.com/aspnet/EntityFrameworkCore/pull/15232

### CODE REVIEW

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md [hunk 1]
```diff
@@ -913,6 +913,34 @@ For example:
 modelBuilder.Entity<Samurai>().HasOne("Some.Entity.Type.Name", null).WithOne();
 ```
 
+## The return type for several async methods has been changed from Task to ValueTask
+
```
Comment (divega) [inline entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Please add the link to the tracking issue here (like the other notes)

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md [hunk 2]
```diff
@@ -913,6 +913,34 @@ For example:
...
+The aforementioned methods now return a `ValueTask<T>` over the same `T` as before.
+
+**Why**
+
+This change reduces the number of heap allocations incurred when invoking these methods, improving general performance.
+
+**Mitigations**
+
+Applications simply awaiting the above APIs only need to be recompiled - no source changes are necessary.
+A more complex usage (e.g. passing the returned `Task` to `Task.WhenAny()`) typically require that the returned `ValueTask<T>` be converted to a `Task<T>` by calling `ToTask()` on it.
```
Comment (divega) [inline entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Is it ToTask or AsTask?

Comment (roji) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Good catch!

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md (hunk 1 above)
Comment (roji) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Added.

**Comment by roji** [comment]
Force-pushed with @divega's corrections, will wait for #15184 to be merged.

**Comment by divega** [comment]
Looks good. I will hold form merging until we know for sure this goes in preview 4 and not 5.

**Comment by roji** [comment]
@divega merged aspnet/EntityFrameworkCore#15232 for preview4, so this should be OK for merging too, I'll let you do the honors.

**Pull request merged**